### PR TITLE
Building workflow improvements

### PR DESCRIPTION
**Pull request by acidicMercury8**
Changes proposed in this pull request:

- Revised building workflows

Agreements

- [x] I agree to follow this project's [__Contributing Guidelines__](https://github.com/hexalang/.github/blob/kawaii/CONTRIBUTING.md)
- [x] I agree to follow this project's [__Code of Conduct__](https://github.com/hexalang/.github/blob/kawaii/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: .github/workflows/build-compiler.yml [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+      - name: Build Hexa
+        run: |
+          node --trace-uncaught bootstrap.js --define debug=false hexa.json
+          node --trace-uncaught hexa-node.js --define debug=true --define times=false hexa.json
+          node --trace-uncaught hexa-node.js --define debug=false --define times=false hexa.json
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: hexa-node.js-${{ matrix.os }}-${{ github.sha }}
```
Comment (PeyTy) [inline .github/workflows/build-compiler.yml:70]: @acidicMercury8 must be xxx-yyy.js (. extension at the end), not xxx.js.yyy, same for other files

File: .github/workflows/build-compiler.yml [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Get Hexa
+        uses: actions/download-artifact@v3
+        with:
+          name: hexa-node.js-${{ matrix.os }}-${{ github.sha }}
```
Comment (PeyTy) [inline .github/workflows/build-compiler.yml:101]: Same

File: .github/workflows/build-compiler.yml (hunk 1 above)
Comment (PeyTy) [reply .github/workflows/build-compiler.yml:70]: @acidicMercury8 does it affect the downloaded file name? Bruh nevermind

**Review by PeyTy — APPROVED** [review]

File: .github/workflows/build-compiler.yml (hunk 1 above)
Comment (acidicMercury8) [reply .github/workflows/build-compiler.yml:70]: @PeyTy `name` in upload step is name of the archive, which contains packed artifacts. This is not the name of the artifact itself. It is recommended to use this to differentiate names

**Pull request merged**